app/services/cart_service.rb

```diff
@@ -17,45 +17,42 @@ def populate(from_hash, overwrite = false)
     # this validation probably can't fail
     if !distribution_can_supply_products_in_cart(@distributor, @order_cycle)
       errors.add(:base, I18n.t(:spree_order_populator_error))
+      return false
     end
 
-    if valid?
-      @order.with_lock do
-        variants = read_variants from_hash
-
-        variants.each do |v|
-          if varies_from_cart(v)
-            attempt_cart_add(v[:variant_id], v[:quantity], v[:max_quantity])
-          end
-        end
-
-        if overwrite
-          variants_removed(variants).each do |id|
-            cart_remove(id)
-          end
-        end
-      end
+    @order.with_lock do
```

Comment:
This is changing logic, not just code style. Why do we take the `valid?` check out?

Reply:
If you look at valid, its just errors.empty?.
I am returning false after the previous if statement that populates errors (line 20). This is the only way valid would be false. So, I don't think there's any logic change.